src/kernels/jupyter/session/jupyterSession.ts

```diff
@@ -240,14 +241,28 @@ export class JupyterSession extends BaseJupyterSession implements IJupyterKernel
         token: CancellationToken;
         ui: IDisplayOptions;
     }): Promise<ISessionWithSocket> {
-        // Create our backing file for the notebook
-        const backingFile = await this.backingFileCreator.createBackingFile(
-            this.resource,
-            this.workingDirectory,
-            this.kernelConnectionMetadata,
-            this.connInfo,
-            this.contentsManager
-        );
+        let backingFile: IBackupFile | undefined;
+        let remoteFilePath: string | undefined;
+
+        if (isRemoteConnection(this.kernelConnectionMetadata) && this.connInfo.workingDirectory && this.resource) {
+            const currentWorkingDirectory = workspace.getWorkspaceFolder(Uri.from(this.resource));
+            if (currentWorkingDirectory) {
+                remoteFilePath = this.resource.path.replace(
+                    currentWorkingDirectory.uri.path,
+                    this.connInfo.workingDirectory
+                );
+            }
+        } else {
+            // Create our backing file for the notebook
+            backingFile = await this.backingFileCreator.createBackingFile(
+                this.resource,
+                this.workingDirectory,
+                this.kernelConnectionMetadata,
+                this.connInfo,
+                this.contentsManager
+            );
+            remoteFilePath = backingFile?.filePath;
+        }
```

Comment:
I'd change the code to ensure we always have a value for `remoteFilepath` which matches closely to the previous code.

```suggestion
        if (isRemoteConnection(this.kernelConnectionMetadata) && this.connInfo.workingDirectory && this.resource) {
            const currentWorkingDirectory = workspace.getWorkspaceFolder(Uri.from(this.resource));
            if (currentWorkingDirectory) {
                remoteFilePath = this.resource.path.replace(
                    currentWorkingDirectory.uri.path,
                    this.connInfo.workingDirectory
                );
            }
        } 
        if (!remoteFilePath) {
            // Create our backing file for the notebook
            backingFile = await this.backingFileCreator.createBackingFile(
                this.resource,
                this.workingDirectory,
                this.kernelConnectionMetadata,
                this.connInfo,
                this.contentsManager
            );
            remoteFilePath = backingFile?.filePath;
        }
```